C/common/config_category.cpp

```
@@ -458,6 +458,164 @@ string ConfigCategory::getValue(const string& name) const
 	throw new ConfigItemNotFound();
 }
 
+/**
+ * Return the value of the configuration category item with a default
+ *
+ * @param name         The name of the configuration item to return
+ * @param defaultValue The default value to return if the item does not exist
+ * @return string      The configuration item value or the default
+ */
+string ConfigCategory::getValue(const std::string& name, const std::string& defaultValue) const
+{
+	try
+	{
+		return getValue(name);
+	}
+	catch (ConfigItemNotFound& e)
+	{
+		Logger::getLogger()->info("'%s' %s , returning default value '%s'", name.c_str(), e.what(), defaultValue.c_str());
+		return defaultValue;
+	}
+	catch(...)
+	{
+		Logger::getLogger()->info("Unkown exception occured returning default value %s", defaultValue.c_str());
+		return defaultValue;
+	}
+}
+
+/**
+ * Return a boolean value from a configuration category item
+ *
+ * @param name         The name of the item
+ * @param defaultValue The value to return if item is not found or invalid
+ * @return bool        The boolean value
+ */
+bool ConfigCategory::getBoolValue(const std::string& name, bool defaultValue) const
+{
+	try
+	{
+		string val = getValue(name);
+		std::string lower = val;
+		std::transform(lower.begin(), lower.end(), lower.begin(), ::tolower);
+		if (lower == "true" || lower == "1") return true;
+		if (lower == "false" || lower == "0") return false;
+		Logger::getLogger()->info("Config item '%s' expected to be boolean but got '%s'", name.c_str(), val.c_str());
```

I think when configItem exist but excepted datatype mismatch then returning default value could be misleading.
@MarkRiddoch  we should raise exception here, caller should know about this condition.

---

This is extremely unlikely to happen as the API for configuration categories will guard against it. We would only end up getting the error if somebody went in and edited the database directly.